add syntax highlight in both edit and preview mode

guopenghui opened the pull request:
Hi, thank you for this useful plugin!

I have enhanced this plugin with some improvements:

![image](https://github.com/runofthemillgeek/swarnam-obsidian/assets/13451866/00097c80-aca2-4b8e-8143-a0c9ba91abb9)
![image](https://github.com/runofthemillgeek/swarnam-obsidian/assets/13451866/5696dd6e-f5ed-4174-800b-27688f4b3540)

guopenghui committed:
feat: add syntax highlight in edit mode

guopenghui committed:
feat: add syntax highlight in preview mode

guopenghui committed:
fix: remove undesirable padding

runofthemillgeek commented:
Thanks for the interest and contribution! :) I didn't have plans to extend this when I started the project, but I think it'd be a nice addition to have something like Prism & CodeMirror (I've personally been exploring Shiki for my Anki setup).

I'm unfamiliar with the Plugin APIs you introduced so I'll try to go over this during the weekend and see if it's all good.

guopenghui commented:
No problem at all, just take a look at it in your free time. I added these features for my own usage needs and thought maybe I could contribute something to the source.

The APIs I introduced are Prism and CodeMirror mode. Prism is pretty simple, with only one `highlight` function included in obsidian.
For codemirror mode, you can refer to:
+ cm5 doc in [this page](https://codemirror.net/5/doc/manual.html#modeapi)
+ [mlir-syntax-highlight plugin code](https://github.com/Lewuathe/obsidian-mlir-syntax-highlight/blob/main/mode/mlir/mlir.js)
+ [obsidian-gdscript code](https://github.com/RobTheFiveNine/obsidian-gdscript/blob/main/src/main.js)

This implementation is just utilizing builtin html/css/js modes.
